common/env/env.go

```diff
@@ -0,0 +1,393 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package env provides a representation of a CEL environment.
+package env
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+
+	"github.com/google/cel-go/common/decls"
+	"github.com/google/cel-go/common/types"
+)
+
+// NewConfig creates an instance of a YAML serializable CEL environment configuration.
+func NewConfig() *Config {
+	return &Config{
+		Imports:    []*Import{},
+		Extensions: []*Extension{},
+		Variables:  []*Variable{},
+		Functions:  []*Function{},
+	}
+}
+
+// Config represents a serializable form of the CEL environment configuration.
+//
+// Note: custom validations, feature flags, and performance tuning parameters are
+// not (yet) considered part of the core CEL environment configuration and should
+// be managed separately until a common convention for such configuration settings
+// can be developed.
+type Config struct {
```

Comment:
Is being able to subset type identifiers desirable?

Reply:
It is, but I haven't figured out how to introduce this concept into the type registries / type providers yet.